sklearn/utils/_testing.py

```diff
@@ -605,6 +609,203 @@ def check_docstring_parameters(func, doc=None, ignore=None):
     return incorrect
 
 
+def _check_item_included(item_name, args):
+    """Helper to check if item should be included in checking."""
+    if args.include is not True and item_name not in args.include:
+        return False
+    if args.exclude is not None and item_name in args.exclude:
+        return False
+    return True
+
+
+def _diff_key(line):
+    """Key for grouping output from `context_diff`."""
+    if line.startswith("  "):
+        return "  "
+    elif line.startswith("- "):
+        return "- "
+    elif line.startswith("+ "):
+        return "+ "
+    elif line.startswith("! "):
+        return "! "
+    return None
+
+
+def _get_diff_msg(str_grouped_dict):
+    """Get message showing the difference between type/desc strings of all objects.
+
+    `str_grouped_dict` keys should be the type/desc strings and values are a list
+    of objects with that string. Objects with the same type/desc string are
+    thus grouped together.
+    """
+    msg_diff = ""
+    ref_str = ""
+    ref_group = []
+    for docstring, group in str_grouped_dict.items():
+        if not ref_str and not ref_group:
+            ref_str += docstring
+            ref_group.extend(group)
+        diff = list(
+            context_diff(
+                ref_str.split(),
+                docstring.split(),
+                fromfile=str(ref_group),
+                tofile=str(group),
+                n=8,
+            )
+        )
+        # Add header
+        msg_diff += "".join((diff[:3]))
+        # Group consecutive 'diff' words to shorten error message
+        for start, group in groupby(diff[3:], key=_diff_key):
+            if start is None:
+                msg_diff += "\n" + "\n".join(group)
+            else:
+                msg_diff += "\n" + start + " ".join(word[2:] for word in group)
+        # Add new lines at end of diff, to separate comparisons
+        msg_diff += "\n\n"
+    return msg_diff
+
+
+def _check_grouped_dict(grouped_dict, type_or_desc, section, n_objects):
+    """Helper to check only one type/desc key in grouped_dict.
+
+    If item is not present in all objects, checking is skipped and warning raised.
+    """
+    skipped = []
+    for item_name, str_grouped_dict in grouped_dict.items():
+        # If item not found in all objects, skip
+        if sum(map(len, str_grouped_dict.values())) < n_objects:
+            skipped.append(item_name)
```

Comment:
I would prefer a list comprehension instead of a `map`:


```suggestion
        if sum([len(item) for item in str_grouped_dict.values()]) < n_objects:
```